includes/optional-modules/nextdoor/class-api.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * Nextdoor API client.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Nextdoor;
+
+use Newspack\Nextdoor;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Nextdoor API client class.
+ */
+class API {
+
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var API
+	 */
+	protected static $instance = null;
+
+	/**
+	 * API base URL.
+	 */
+	const API_BASE_URL = 'https://nextdoor.com';
+
+	/**
+	 * Main API Instance.
+	 *
+	 * @return API - Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Make API request.
+	 *
+	 * @param string $endpoint API endpoint.
+	 * @param array  $args Request arguments.
+	 * @param string $method HTTP method.
+	 * @return array|WP_Error
+	 */
+	private function make_request( $endpoint, $args = [], $method = 'GET' ) {
+		$url = self::API_BASE_URL . $endpoint;
```

Comment:
```suggestion
		$url = trailingslashit( self::API_BASE_URL ) . $endpoint;
```
Just so we don't accidentally end up with `https://nextdoor.compartner/v1/` :)

Reply:
Right!, Added this normalization ✅